web/pages/article/[slug].tsx

```diff
@@ -14,7 +14,16 @@ import MetaTags from '../../components/MetaTags';
 
 const QUERY = groq`
   {
-    "article": *[_type == "article" && slug.current == $slug][0],
+    "article": *[_type == "article" && slug.current == $slug][0] {
+      ...,
+      content[] {
+        ...,
+        markDefs[] {
+          ...,
+          reference->,
```

Comment:
When I try this query in Vision it seems to fetch lots more content than needed, e.g. the entire rich text contents of the article in the "Link 2" case. I would think that this means the client-side JSON payload is that much larger if you navigate to the article (that contains the internalLink) via Next.js `<Link>` navigation?

Can this be limited to only the fields we actually need? Which I think would be these:

```suggestion
          reference->{_type, slug},
```

Reply:
Good catch! 7c9e547.